protoc-gen-grpc-gateway/internal/gengateway/template.go

```diff
@@ -72,6 +72,9 @@ func (b binding) QueryParamFilter() queryParamFilter {
 	}
 	for _, p := range b.PathParams {
 		seqs = append(seqs, strings.Split(p.FieldPath.String(), "."))
+		if p.Target.JsonName != nil {
+			seqs = append(seqs, strings.Split(*p.Target.JsonName, "."))
+		}
```

Comment:
We probably also want to do this for the Body parameters as well, since the same thing could happen there, I think? Could you try and see what happens if you try to override a body parameter?